src/voodoo/VoodooClient.ts

```diff
@@ -215,8 +194,9 @@ export default class VoodooClient {
 
   private async ensureUserContactPublished(): Promise<void> {
     const bundle = await this.getUserContactFromNetwork(this.address)
-    // TODO: validate the bundle, not just check for presence
-    if (bundle) {
+    // NOTE: other devices for this wallet could have published bundles, this is expected
+    // TODO: we only avoid republishing our own bundle if the account is the same
+    if (!!bundle && bundle.voodooInstance === this.voodooInstance) {
```

Comment:
This equality check is only validating whether the two instances point to the same memory address, not a deep equality of whether they are equivalent. 

So, it will never be true.

Reply:
Ah good catch

Reply:
I'll fix in the follow up PR